community/becoming-a-maintainer.md

```diff
@@ -1,3 +1,77 @@
 # Becoming a maintainer
 
-TODO
+## Introduction
+
+Our Maintainers are the people within Exercism that manage the different parts of our ecosystem. 
+We aim to have at least active two maintainers for each programming language track, and at least one active maintainer for all the parts of the platform.
+
+We have a formalised process for becoming and staying a maintainer, to make it achievable and fair for anyone.
+**You can be become a maintainer!**
+Read on to understand the responsibilities and the process.
+
+_Note: the second below generally discussses Track Maintainers. 
+The process for becoming a maintainer of a different part of the ecosystem is similar but a little looser._
+
+## Responsibilities
+
+Maintainers are responsible for ensuring that a Track works well within Exercism, and continues to grow and develop. 
+That specifically means that maintainers:
+- **Set technical direction for a track:** Choosing the right language versions, tooling, etc.
+- **Manage the syllabus of a track:** Designing, choosing and defining the right Concepts and Exercises for a Track, and ensuring they are linked and structured appropriately.
+- **Ensure a traack track work well:** Ensuring the Track is up to date, that its test-runner is fast, etc.
+- **Ensure the track conforms to Exercism's standard:** Exercism has specifications and standards for each language Track to ensure a degree of consistency within the platform. While maintainers have free reign over much of the Track, all Tracks must conform to Exercism's standards.
+
+Maintainers are also responsibile for interacting with our contributor community.
+They review PRs from community, acting as gatekeepers, shephards, and cheerleaders for our community.
+They are responsible for keeping documentation and contribution guidelines up to date, and ensuring that there are relevant open issues for potential contributors to action.
+
+It is essential that Maintainers manifest Exercism's values of empathy and kindness, while maintaining high standards for the Track and the platform.
+
+## Powers
+
+The primary way that Maintainers are able to their enact their responsibilities is by having the write access to repositories. 
+Through this Maintainers control what code goes into their repositories.
+
+## Rewards
+
+The main incentives for being an Exercism Maintainer are intrinsic: working on a meaningful project; working with other smart, kind people; developing both technical and soft skills; having a meaningful out-of-work entry on your résumé.
+
+We also provide some extra benefits in:
+- A "Maintainer" badge on profiles and around the website.
+- Blog posts and interviews with maintainers.
+- Exclusive Maintainer swag
+
+## Maintainer requirements 
+
+Maintainers commit themselves to spending an average of ~4 hours per week on maintaining their Track(s).
```

Comment:
"Requirements" sounds maybe too much? Maybe this is more about "Achieving maintainer status"?

4 hours per week is too much (or maybe just too consistent?) for me to commit to. I've got a demanding job that bleeds outside office hours, and a number of other side projects, and will occasionally take a week or two off to focus on those.

Also, at least for the "smaller" languages, much smaller time investments can accomplish a lot. (So, maybe when the opportunity is high, at least, these requirements should be relaxed)

As it's worded now, if I came across this I probably would not have considered contributing to exercism tracks. Maybe I'd word it something like this:

```
## Achieving maintainer status

All contributions are gladly welcome. "Exercism Maintainer" status is intended for people who are
regularly involved. (See "Path to becoming a Maintainer" below)

The general expectation for maintainers is ~4 hours per week on average of maintenance effort.
Some communities may grant maintainer status at different levels of involvement.
```

Not saying "use this exact wording" cause it's off-the-cuff and a little clunky, (maybe just needs to be organized?) but just thinking that more flesh on the bones here can make this a little less scary before going into the nitty-gritty details below

Reply:
> 4 hours per week is too much (or maybe just too consistent?) for me to commit to. I've got a demanding job that bleeds outside office hours, and a number of other side projects, and will occasionally take a week or two off to focus on those.

I don't think it is too much to state as a formal requirement. It's below what many if not most volunteering projects require as commitment. Considering that this is averaged over an unspecified time (6 months?), and you already get ~1.5h/week from merely attending the calls to keep up-to-date on what's going on, ~4h/w seem more than fair to me. Setting it too low might result in people accepting the role as maintainer when a normal contributor role might fit them better.

> As it's worded now, if I came across this I probably would not have considered contributing to exercism tracks.

Perhaps it should be clarified that becoming a maintainer should not be seen as something "better" than contributor?

Reply:
>As it's worded now, if I came across this I probably would not have considered contributing to exercism tracks. 

Yeah, so as you go on to say in your example, you can contribute without being a maintainer though. In other large projects its normal to contribute for a year or two before getting maintainer privileges. Giving someone write-access to our repos is something we should protect I think.

I also think that for unmaintained (and especially unlaunched tracks such as Factor) the barrier is much lower. Getting 50 hours from someone over a year is wonderful and I might well give them maintainer rights for that, just so they can be autonomous. But still in general, if someone is only able to spend an hour a week, that's probably at most one PR, which the Reviewer Maintainers can check without that contributor needing to have that write-access.

I think we're all saying the same thing here really. I like your suggestion - I'll tweak and add it tomorrow.

Reply:
@hiljusti 
> not have considered contributing to exercism tracks.

Not considered contributing because of a maintainer threshold?  Or did you mean to say you may not have considered becoming a maintainer?  Contributor threshold is much lower than this, and it would be unfortunate to not contribute because of a different thing.

Reply:
> I think it's perfectly fine for a maintainer to do less than that in practice if there's a good reason but the up-front commitment should be sufficiently high.

I agree. We've just had so many maintainers that did not appreciate that maintaining a track does require quite a bit of effort. Preventing that from happening is important, as we want to be able to accurately communicate that a track is being actively maintained.

> Yeah, so as you go on to say in your example, you can contribute without being a maintainer though.

This 100%. We've had lots of this happening, where people start contributing and were then asked to become a maintainer later. At this point, having _this_ document becomes very important to get people to understand what it means to be a maintainer.

In other words, one of the goals of this document to me is to be able to point it to a potential maintainer and have that person be able to make an informed decision on whether that person wants to be a maintainer.

Reply:
I've rewritten the intro to incorporate these comments.

Reply:
To @kotp question, I think what I was trying to articulate was it sounded like "please make a time commitment" and missed an idea of "all contributions welcome"

Granted, I kind of skimmed the doc out of order, I think the idea is there if you have better reading habits than me

Reply:
@hiljusti is the final version clearer? https://exercism.lol/docs/community/maintainers